### ISSUE TITLE
signin with email and password

### ISSUE THREAD

**Issue by rodgomesc** [convergencelabs/convergence-project#174]
**Versions**
*Please fill in all that apply:*
 - Convergence Version: **1.0.0-rc.7**
 - OS: **Linux**
 - Browser: **all**
 - Java Version: **openjdk 11**

**Description**

hello, guys, I didn't see anything in the documentation related to login with email and password, only with username, jwt, or anonymously, would it be possible to do this implementation? or if you can show me some methods responsible for the login, and their flows, I could try to implement, thanks

**Comment by mmacfadden** [comment]
You are correct, this feature does not exist. I think this would be pretty easy to add. We will add this shortly.

**Comment by rodgomesc** [comment]
thank you, i will implement this project in a service that i am developing, so i must be more familiar with the convergence code sun, and i will be able to contribute, as soon as i can contribute i will start implementing a client in python and ruby, from what i have seen so far it didn't seem so complicated

**Comment by mmacfadden** [comment]
The client side is not too complicated.  We would love to collaborate with you.

**Comment by mmacfadden** [comment]
We are working on this now. This will likely involve a small breaking API change.  RIght now we have the following method:

```
public connectWithPassword(credentials: IUsernameAndPassword | (() => Promise<IUsernameAndPassword>)): Promise<void> 
```

And `IUsernameAndPassword` looks like:

```
export interface IUsernameAndPassword {
  username: string;
  password: string;
}
```


We will update this to be:


```
export interface IPasswordCredential {
  username?: string;
  email?: string;
  password: string;
}
```

This will allow the caller to provide an email or username.  The Interface name will change though, to correctly name the interface. Not a big change, but it will be breaking.

**Comment by rodgomesc** [comment]
sounds good,

when ready i will test it, thanks

**Comment by rodgomesc** [comment]
just so as not to confuse other people to come and read this issue,

I think you meant ...

This will allow the caller to provide an **email** or **username**

**Comment by mmacfadden** [comment]
One thing I did find, looking at the domain schema it appears that email is not a unique constraint for domain users. I would think we would need to make domain user emails unique if this were going to work.
